### ISSUE TITLE
Documentation Request: How to specify predefined timestamps

### ISSUE THREAD

**Issue by visualcurrent** [mblackstock/node-red-contrib-influxdb#110]
**TL;DR** Please tell us how to include our event time (as opposed to server ingestion time) in each Output Node Usage Notes for each payload style.

InfluxDB clearly defines in their [Line Protocol](https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/) that this will enter data with the timestamp we provide

**measurement**`<comma>`**tagset**`<space>`**fieldset**`<space>`**timestamp**

It also tells us that the timestamp is optional. Leaving it out will record the server based ingestion timestamp.

**measurement**`<comma>`**tagset**`<space>`**fieldset**

The usage notes here don't provide clear documentation for how to override the ingestion time with our own event time.

Just like InfuxDB clearly documents their example with a timestamp and then add that it can be optionally removed to use server time, it would be really great if every Output Node example here would follow this useful convention. 

Please show or describe how predefined timestamps are included in the various payload styles.

-  If the **msg.payload** is a string, number, or boolean
    - I expect that it can't be done in this style. But if it can, please divulge.
- If **msg.payload** is an object containing multiple properties
    - Is there a specially reserved key name for timestamp that can be added to the object?
- If **msg.payload** is an array containing two objects
    - Where would a timestamp go? A third object?
- if **msg.payload** is an array of arrays
    - Same... a third object?
- The Batch Output Node
    - `timestamp: new Date()` is shown here. What's important? Does the key name matter? Is this key/value optional?

**Comment by TheNON75** [comment]
Hi there,

i hope the code is still maintained.
I have an issue with the timestamp as well and any help is welcome.
In case using `time: new Date()`, according to the influx api spec, it is ignored.
However, when using `timestamp: new Date()` or `timestamp: Date.now()` results in nothing useful.

May someone point me to the right direction how to use own/predefined/custom/static/whatever timestamp?

thanks
regards

**Comment by mblackstock** [comment]
Thanks for your feedback on the documentation.  Apologies if it is not clear.  I'd be happy to review a pull request if you would like to improve it.

If there is a bug, that is, the `time` or `timestamp` properties are not working as you'd expect, please open another issue and I will have a look.  Again, a pull request for any fixes are welcome.

To specify the time using the `output` node, you use the `time` reserved property in any of the payload formats as shown in the last example.

Unlike the output node, the `batch` node uses a `timestamp` property that is at the same level as the measurement, tags and fields properties.

**Comment by TheNON75** [comment]
> 

thank you, i've created a new issue and make a pr afterwards if necessary
[](https://github.com/mblackstock/node-red-contrib-influxdb/issues/112#issue-1123985827)
